### PR TITLE
remove VMDisksFirehose component in favor of a hook

### PR DESCRIPTION
**Pull request by suomiy**

### CODE REVIEW

**Comment by spadgett** [comment]
/override ci/prow/ceph-storage-plugin

The ceph-storage-plugin job should only have run on PRs with changes to `frontend/packages/ceph-storage-plugin`. It failed due to an unrelated CI config change. Overriding status on this PR.

File: frontend/packages/kubevirt-plugin/src/components/vm-disks/vm-disks.tsx [hunk 1]
```diff
@@ -227,47 +232,18 @@ export const VMDisks: React.FC<VMDisksProps> = ({ vmLikeEntity, vmTemplate, vmi
...
-  const resources = [
-    getResource(TemplateModel, {
-      name: vmTemplate?.name,
-      namespace: vmTemplate?.namespace,
-      isList: false,
-      prop: 'vmTemplate',
-    }),
-  ];
-
+export const VMDisksAndFileSystemsPage: React.FC<VMTabProps> = (props) => {
```
Comment (metalice) [inline frontend/packages/kubevirt-plugin/src/components/vm-disks/vm-disks.tsx:235]: please consider making this nested component in his own file

File: frontend/packages/kubevirt-plugin/src/components/vm-disks/vm-disks.tsx [hunk 2]
```diff
@@ -151,20 +152,24 @@ export const VMDisksTable: React.FC<React.ComponentProps<typeof Table> | VMDisks
...
-  vmLikeEntity?: VMGenericLikeEntityKind;
-  pvcs?: FirehoseResult<K8sResourceKind[]>;
-  datavolumes?: FirehoseResult<V1alpha1DataVolume[]>;
-  vmTemplate?: FirehoseResult<TemplateKind>;
+type VMDisksProps = VMTabProps & {
   vmi?: VMIKind;
 };
 
-export const VMDisks: React.FC<VMDisksProps> = ({ vmLikeEntity, vmTemplate, vmi }) => {
+export const VMDisks: React.FC<VMDisksProps> = ({ obj: vmLikeEntity, vmi }) => {
```
Comment (metalice) [inline frontend/packages/kubevirt-plugin/src/components/vm-disks/vm-disks.tsx:159]: please consider making this nested component in his own file

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/vm-disks/vm-disks.tsx:159]: IMO this is a bit out of scope of this PR. I suppose that we could split this file into more files, but that would get lost in the current folder which has 10 files already.

We probably need a better structure and larger refactoring to tackle other components in this folder as well.

File: frontend/packages/kubevirt-plugin/src/components/vm-disks/vm-disks.tsx (hunk 1 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/vm-disks/vm-disks.tsx:235]: https://github.com/openshift/console/pull/7879#discussion_r595350356

**Comment by suomiy** [comment]
reclasify as a bug

**Pull request closed without merging**